ledger/ledger-consensual/src/main/kotlin/net/corda/v5/ledger/consensual/transaction/ConsensualLedgerTransaction.kt

```diff
@@ -0,0 +1,16 @@
+package net.corda.v5.ledger.consensual.transaction
+
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.ledger.common.transactions.LedgerTransaction
+import net.corda.v5.ledger.consensual.ConsensualState
+import net.corda.v5.ledger.consensual.Party
+import java.time.Instant
+
+@DoNotImplement
+interface ConsensualLedgerTransaction: LedgerTransaction {
+    val metadata: ConsensualTransactionMetaData
+    val timestamp: Instant
+    val requiredSigners: List<Party>
```

Comment:
Continuing the discussion of Party. For instance here we talk about signers but include a Party not a PublicKey, or hash of a PublicKey. The signature will only confirm the PublicKey it won't help with the name. Worse the name/key association is likely to shift with time as keys are rotated. It also fails to include account information, which might be needed. So personally I would like to make the list of signers just a list of PublicKey. However, people do require a way of categorising state/transactions by the network members involved and they also need to be able to validate notaries and issuers against MemberInfo details.

Reply:
I have moved this comment here: https://r3-cev.atlassian.net/browse/CORE-5936